addons/purchase_requisition/models/purchase.py

```diff
@@ -212,34 +212,46 @@ def action_compare_alternative_lines(self):
         }
 
     def get_tender_best_lines(self):
-        product_to_best_price_line = defaultdict(lambda: self.env['purchase.order.line'])
-        product_to_best_date_line = defaultdict(lambda: self.env['purchase.order.line'])
-        product_to_best_price_unit = defaultdict(lambda: self.env['purchase.order.line'])
+        product_to_best_price_line = defaultdict(lambda: [self.env['purchase.order.line'], False])
+        product_to_best_date_line = defaultdict(lambda: [self.env['purchase.order.line'], False])
+        product_to_best_price_unit = defaultdict(lambda: [self.env['purchase.order.line'], False])
+        date = self._context.get('date') or fields.Date.today()
+        company = self.env['res.company'].browse(self._context.get('company_id')) or self.env.company
```

Comment:
Do we need the `self.env['res.company'].browse(self._context.get('company_id'))`?